File: 2.py
import time

bsize = int(input('Bucket size: '))
orate = int(input('Ouptut rate: '))
store = 0
print()

while(True):
    inc = int(input('Incoming packet size(<0 to stop): '))
    if(inc < 0):
        quit()
    if(store + inc > bsize):
        print('Packet dropped since bucket size exceeded')
    else:
        store += inc
    print('Transimission left: ', store)
    nex = int(input('Next packet will come in: '))
    print()
    for i in range(nex):
        time.sleep(1)
        if(store):
            if(store > orate):
                print('Transmitted packet of size: ', orate)
                store -= orate
            else:
                print('Transmitted packet of size: ', store)
                store = 0
            print('Transmission left: ', store)
        else:
            print('No packets left to trasmit')
        print('Time left: ', nex - i - 1)
        print()

File: 2.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
	int i, b_size, o_rate, n, temp, cur_b=0;
	vector<int>p;
	cout<<"Enter bucket size: ";
	cin>>b_size;
	cout<<"Enter output rate: ";
	cin>>o_rate;
	cout<<"Enter number of packets: ";
	cin>>n;
	cout<<"Enter the packets: ";
	for(i=0; i<n; i++)
	{
		cin>>temp;
		p.push_back(temp);
	}
	for(i=0; i<n; i++)
	{
		if(p[i]+cur_b<=b_size+o_rate)
		{
			cur_b=cur_b+p[i];
			cout<<"Packet "<<i<<" transmitted! ";
		}
		else
			cout<<"Packet "<<i<<" discarded! ";
		cur_b=cur_b-o_rate;
		if(cur_b<0)
			cur_b=0;
		cout<<"Bucket status "<<cur_b<<"/"<<b_size<<endl;
	}
	return 0;
}